File: INTRODUCCION-POO/EJERCICIO-3/Coche.py
class Coche:
    def __init__(self, marca, modelo, velocidad=0):
        self.marca = marca
        self.modelo = modelo
        self.velocidad = velocidad

    def acelerar(self):
        self.velocidad += 10

    def frenar(self):
        if self.velocidad >= 5:
            self.velocidad -= 5
        else:
            self.velocidad = 0

    def mostrar_velocidad(self):
        print(f"El coche {self.marca} {self.modelo} va a una velocidad de {self.velocidad} km/h.")

# Crear dos coches
coche1 = Coche("Toyota", "Corolla")
coche2 = Coche("Volkswagen", "Golf")

# Acelerar varias veces
coche1.acelerar()
coche1.acelerar()
coche2.acelerar()
coche2.acelerar()
coche2.acelerar()

# Mostrar las velocidades 
print("Velocidades después de acelerar:")
coche1.mostrar_velocidad()
coche2.mostrar_velocidad()

# Frenar varias veces
coche1.frenar()
coche2.frenar()
coche2.frenar()
coche2.frenar()
coche2.frenar() # Intentar frenar más de lo posible

# Mostrar las velocidades después de frenar
print("\nVelocidades después de frenar:")
coche1.mostrar_velocidad()
coche2.mostrar_velocidad()

File: INTRODUCCION-POO/EJERCICIO-3/Coche.java
public class Coche {
    String marca;
    String modelo;
    int velocidad;

    public Coche(String marca, String modelo) {
        this.marca = marca;
        this.modelo = modelo;
        this.velocidad = 0;
    }

    public void acelerar() {
        this.velocidad += 10;
    }

    public void frenar() {
        if (this.velocidad >= 5) {
            this.velocidad -= 5;
        } else {
            this.velocidad = 0;
        }
    }

    public void mostrarVelocidad() {
        System.out.println("El coche " + this.marca + " " + this.modelo + " va a una velocidad de " + this.velocidad + " km/h.");
    }

    public static void main(String[] args) {
        Coche coche1 = new Coche("Toyota", "Corolla");
        Coche coche2 = new Coche("Volkswagen", "Golf");

        coche1.acelerar();
        coche1.acelerar();
        coche2.acelerar();
        coche2.acelerar();
        coche2.acelerar();

        System.out.println("Velocidades después de acelerar:");
        coche1.mostrarVelocidad();
        coche2.mostrarVelocidad();

        coche1.frenar();
        coche2.frenar();
        coche2.frenar();
        coche2.frenar();
        coche2.frenar(); 

        System.out.println("\nVelocidades después de frenar:");
        coche1.mostrarVelocidad();
        coche2.mostrarVelocidad();
    }
}
